alerting/src/main/kotlin/org/opensearch/alerting/MonitorRunner.kt

```diff
@@ -757,39 +759,119 @@ object MonitorRunner : JobRunner, CoroutineScope, AbstractLifecycleComponent() {
             return
         }
 
+        val count: Int = lastRunContext["shards_count"] as Int
+        val updatedLastRunContext = lastRunContext.toMutableMap()
```

Comment:
Quick clarifying question. Is this `toMutableMap` call redundant? I see that we also call it on line [752](https://github.com/opensearch-project/alerting/pull/316/files#diff-1afa0c5b81f1bd6459dc8398dd5f24cc989b9fa3447541519a50dc5a50e06286R752), and line [755](https://github.com/opensearch-project/alerting/pull/316/files#diff-1afa0c5b81f1bd6459dc8398dd5f24cc989b9fa3447541519a50dc5a50e06286R755).

Reply:
That is so a new map object is created and is not a reference to the old object.

Reply:
Ah, good to know.